pkg/apis/serving/v1beta1/predictor_custom.go

```diff
@@ -65,8 +65,12 @@ func (c *CustomPredictor) Default(config *InferenceServicesConfig) {
 	if len(c.Containers) == 0 {
 		c.Containers = append(c.Containers, v1.Container{})
 	}
-	c.Containers[0].Name = constants.InferenceServiceContainerName
-	setResourceRequirementDefaults(&c.Containers[0].Resources)
+	for i, _ := range c.Containers {
+		if c.Containers[i].Name == "" {
+			c.Containers[i].Name = constants.InferenceServiceContainerName
```

Comment:
If Containers are not be set the name, then they will be set the same container name. I suggest adding the index of `c.Containers` behind the `constants.InferenceServiceContainerName`